### PR TITLE
Add 'type' back to transfers dict for retry job

### PR DESCRIPTION
**Pull request by novicecpp**
Fix https://github.com/dmwm/CRABServer/issues/8005

Add `type` back for transfer dicts of second retry. Also pop out some unnecessary key.

### CODE REVIEW

File: src/python/TaskWorker/Actions/PostJob.py [hunk 1]
```diff
@@ -956,6 +956,13 @@ def updateOrInsertDoc(self, doc, toTransfer):
                 msg += " Transfer submission failed."
                 msg += "\n%s" % (str(hte.headers))
                 returnMsg['error'] = msg
+            # add/remove some keys from newDoc to has the same schema as `if not self.found_doc_in_db` is True.
```
Comment (belforte) [inline src/python/TaskWorker/Actions/PostJob.py:959]: why doing it here and not in the definition of newDoc a few lines above ?

**Review by belforte — CHANGES_REQUESTED** [review]

File: src/python/TaskWorker/Actions/PostJob.py (hunk 1 above)
Comment (novicecpp) [reply src/python/TaskWorker/Actions/PostJob.py:959]: You are right.

Add `type` back should be fine and nothing bad for have extra key in transfers.txt.
I need to test to make sure REST is working properly.

Comment (novicecpp) [reply src/python/TaskWorker/Actions/PostJob.py:959]: Unfortunately, I need to add extra validation for that https://github.com/dmwm/CRABServer/compare/master...novicecpp:CRABServer:add_type_back_to_transfer_dicts_prcompare?expand=1

In my opinion, `newDoc` should construct before if-else, then you copy the object and manipulate it as you like for talking to REST. But it is a big change. So, I am still leaning toward my original PR here.

What do you think?

Comment (belforte) [reply src/python/TaskWorker/Actions/PostJob.py:959]: I am too tired to understand. Can we have a short chat tomorrow ?

Comment (novicecpp) [reply src/python/TaskWorker/Actions/PostJob.py:959]: Sure.

**Comment by novicecpp** [comment]
Ahhhh, we have already done this for other keys in line 973-983,  before we dump it into `transfers.txt`.
 https://github.com/dmwm/CRABServer/blob/372e3856e4431a8e54c16469d7c7060cf403fdb5/src/python/TaskWorker/Actions/PostJob.py#L973-L986

I will add this below the line 983 instead.

**Review by belforte — APPROVED** [review]

**Comment by belforte** [comment]
very good ! yeah... it is not clear at all what `toTransfer` argument mean here :-(

**Comment by novicecpp** [comment]
Thanks. Merging...

**Pull request merged**